uvicorn/lifespan/on.py

```diff
@@ -43,28 +43,41 @@ def __init__(self, config):
             scope = {"type": "lifespan"}
             await app(scope, self.receive, self.send)
         except BaseException as exc:
+            self.asgi = None
```

Comment:
@tomchristie Any reason that this particular line exists? I grep'ed the code base and couldn't find a `lifespan.asgi` being used anywhere. Since I was unsure I kept it, but maybe it's an ancient artefact that should be removed?